app/src/main/java/com/example/rickandmortyapp/ui/character/CharacterViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.rickandmortyapp.ui.character
```

Comment:
This is the CharacterViewModel, where the main business logic for populating and handling the UI should be concentrated.

I expect you to take care of the loading and errors feedback, I have no intentions to do UI improvements, only show you the proper way to do things. The `TODOs`added as comments in the code are highlighting this.

Notice that the ViewModel is also using Hilt, it receives the SavedState as a plug-and-play behavior from Hilt with the Android environment and architecture, but it's also receiving the CharacterRepository.